src/Fmt_ast.ml

```diff
@@ -348,7 +357,20 @@ let fmt_constant c ~loc ?epi const =
           | Some s -> (s, `Preserve) )
         | _ -> (s, c.conf.escape_strings)
       in
+      let contains_pp_commands =
+        let is_substring substring = String.is_substring s ~substring in
+        List.exists ["@,"; "@;"; "@\\n"] ~f:is_substring
+      in
       match c.conf.break_string_literals with
+      | `Newlines when contains_pp_commands ->
+          let break_on_pp_commands in_ delim =
+            let pat = String.Search_pattern.create delim in
+            let with_ = delim ^ "\n" in
+            String.Search_pattern.replace_all pat ~in_ ~with_
+          in
+          List.fold_left ["@,"; "@;"; "@\n"] ~init:s ~f:break_on_pp_commands
```

Comment:
The 3rd one is not the same as in `contains_pp_commands`: `"@\n"` vs `"@\\n"`. Is it intentional ?

Reply:
I simplified the lists when it was possible, but didn't find a way to use the same lists.